capsules/src/spi.rs

```diff
@@ -289,3 +310,289 @@ impl<'a, S: SpiMasterDevice> SpiMasterClient for Spi<'a, S> {
+                        app_write: None,
+                        len: 0,
+                        index: 0,
+                    },
+                    Some(mut appc) => {
+                        appc.callback = Some(callback);
+                        appc
+                    }
+                };
+                self.app.replace(appc);
+                ReturnCode::SUCCESS
+            },
+            1 /* chip selected */ => {
+                let appc = match self.app.take() {
+                    None => SlaveApp {
+                        callback: None,
+                        selected_callback: Some(callback),
+                        app_read: None,
+                        app_write: None,
+                        len: 0,
+                        index: 0,
+                    },
+                    Some(mut appc) => {
+                        appc.selected_callback = Some(callback);
+                        appc
+                    }
+                };
+                self.app.replace(appc);
+                ReturnCode::SUCCESS
+            },
+            _ => ReturnCode::ENOSUPPORT
+        }
+    }
+    // 0: check if present
+    // 1: read/write a single byte (blocking)
+    //   - No longer supported
+    // 2: read/write buffers 
+    //   - read and write buffers optional
+    //   - fails if arg1 (bytes to write) >
+    //     write_buffer.len()
+    // 3: set chip select
+    //   - not supported in slave mode
+    // 4: get chip select
+    //   - returns current selected peripheral
+    //   - in slave mode, always returns 0
+    // 5: set rate on current peripheral
+    //   - not supported in slave mode
+    // 6: get rate on current peripheral
+    //   - not supported in slave mode
+    // 7: set clock phase on current peripheral
+    //   - 0 is sample leading
+    //   - non-zero is sample trailing
+    // 8: get clock phase on current peripheral
+    //   - 0 is sample leading
+    //   - non-zero is sample trailing
+    // 9: set clock polarity on current peripheral
+    //   - 0 is idle low
+    //   - non-zero is idle high
+    // 10: get clock polarity on current peripheral
+    //   - 0 is idle low
+    //   - non-zero is idle high
+    //
+    // 11: hold CS line low between transfers
+    //   - set CSAAT bit of control register
+    //   - not supported for slave 
+    // 12: release CS line (high) between transfers
+    //   - clear CSAAT bit of control register
+    //   - not supported for slave
+    //
+    // x: lock spi
+    //   - if you perform an operation without the lock,
+    //     it implicitly acquires the lock before the
+    //     operation and releases it after
+    //   - while an app holds the lock no other app can issue
+    //     operations on SPI (they are buffered)
+    //   - not implemented or currently supported
+    // x+1: unlock spi
+    //   - does nothing if lock not held
+    //   - not implemented or currently supported
+    //
```

Comment:
Really great comment, but three slashes (`///`) instead of two will make it show up in rustdoc too.